src/matchers/element/toHaveText.ts

```diff
@@ -10,18 +10,19 @@ import {
 
 async function condition(el: WebdriverIO.Element | WebdriverIO.ElementArray, text: string | RegExp | Array<string | RegExp> | ExpectWebdriverIO.PartialMatcher | Array<string | RegExp>, options: ExpectWebdriverIO.StringOptions) {
     const actualTextArray = []
+    const resultArray = []
     let actualText, checkAllValuesMatchCondition
 
     if(Array.isArray(el)){
-        const checkPromises = el.map(async (value) => {
-            actualText = await value.getText()
+        for(const element of el){
+            actualText = await element.getText()
```

Comment:
We can just define it within this scope (same for the else block)

```suggestion
            const actualText = await element.getText()
```

Reply:
Gotcha :D